client/landing/stepper/declarative-flow/internals/steps-repository/readymade-template-generate-content/index.tsx

```diff
@@ -1,23 +1,138 @@
+import { ONBOARD_STORE, SITE_STORE } from 'calypso/landing/stepper/stores';
+import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
+import { ReadymadeTemplate } from 'calypso/my-sites/patterns/types';
+import generateAIContentForTemplate from './api/generate-content';
+import ReadymadeTemplatePreview from './components/readymade-template-preview';
+import TextProgressBar from './components/text-progress-bar';
+import type { OnboardSelect } from '@automattic/data-stores';
+import './style.scss';
+
+type ReadymadeTemplateGenerateContentProps = {
+	readymadeTemplate: ReadymadeTemplate;
+	siteId: number;
+	siteSlug: string;
+};
+
+const ReadymadeTemplateGenerateContent: React.FC< ReadymadeTemplateGenerateContentProps > = ( {
+	readymadeTemplate,
+	siteId,
+	siteSlug,
+} ) => {
+	const [ aiContext, setAiContext ] = useState( '' );
+	const [ numberOfGenerations, setNumberOfGenerations ] = useState( 0 );
+	const [ isGeneratingContent, setIsGeneratingContent ] = useState( false );
+	const { assembleSite } = useDispatch( SITE_STORE );
+
+	const handleTextareaChange = ( event: { target: { value: string } } ) => {
+		setAiContext( event.target.value );
+	};
+
+	const generateContent = () => {
+		setIsGeneratingContent( true );
+		setNumberOfGenerations( numberOfGenerations + 1 );
+		generateAIContentForTemplate( readymadeTemplate, aiContext )
+			.then( ( rt: ReadymadeTemplate ) =>
+				assembleSite( siteId, 'assembler', {
+					homeHtml: rt.home.content,
+					headerHtml: rt.home.header,
+					footerHtml: rt.home.footer,
+					pages: [],
+					globalStyles: rt.globalStyles,
+					canReplaceContent: true,
+					siteSetupOption: 'readymade-template',
+				} )
+			)
+			.then( () => setIsGeneratingContent( false ) );
+	};
 
-const ReadymadeTemplateGenerateContent = () => {
 	return (
-		<div>
-			<h1>Generate Content with Me</h1>
-			<p>Launchpad things here for reasons</p>
-		</div>
+		<>
+			<div className="generate-content">
+				<DocumentHead title={ translate( 'Customize Content With AI' ) } />
+				<form className="generate-content__form">
+					<FormFieldset className="generate-content__form-fieldset">
+						<FormLabel htmlFor="tagline">{ translate( 'Describe your site' ) }</FormLabel>
+						<p className="generate-content__description">
+							{ translate(
+								`Describe your site in a few sentences. The more details you give us the better results you'll get.`
+							) }
+						</p>
+						<FormTextarea
+							name="tagline"
+							id="tagline"
+							placeholder={ translate( 'Write an amazing description of your site.' ) }
+							onChange={ handleTextareaChange }
+						/>
+						<TextProgressBar target={ 256 } text={ aiContext } />
+					</FormFieldset>
+					<div className="generate-content__buttons">
+						<Button
+							className="checklist-item__checklist-primary-button"
+							onClick={ generateContent }
+							disabled={ numberOfGenerations >= 5 }
```

Comment:
Can we disable the button while the content is being generated to avoid accidental clicks?

Reply:
Sure, the button is now disabled while generating content.